### PR TITLE
Updating GFPT and plotter

### PR DESCRIPTION
**Pull request by SergioMartin86**
A few improvements to the RL plotter and some of the algos

### CODE REVIEW

File: source/modules/solver/agent/agent._cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ void Agent::trainingGeneration()
         _sessionPolicyUpdateCount++;
 
         // If using REFER, updating its information
-        if (_experienceReplayREFEREnabled == true) updateREFERInformation();
```
Comment (webepasc) [inline source/modules/solver/agent/agent._cpp:181]: why no more check?

File: source/modules/solver/agent/continuous/GFPT/GFPT._cpp [hunk 2]
```diff
@@ -245,50 +246,47 @@ void GFPT::trainPolicy()
...
+    std::vector<float> policyTarget(2 * _problem->_actionVectorSize, 0.0f);
+
+    // Getting REFER gradients
     auto referKLDivGrads = calculateKLDivergenceGradient(curMeans, curSigmas, expMeans, expSigmas);
 
-    // Storage for the policy gradients
-    std::vector<float> policyGradients(2 * _problem->_actionVectorSize, 0.0f);
+    // Re-computing importance weight for the experiment and storing it into cache
+    float importanceWeight = calculateImportanceWeight(expAction, curMeans, curSigmas, expMeans, expSigmas);
+    _experienceReplay[expId].cache.set("Importance Weight", importanceWeight);
```
Comment (webepasc) [inline source/modules/solver/agent/continuous/GFPT/GFPT._cpp:257]: why is the importance weight here still cached?

**Review by webepasc — APPROVED** [review]

File: source/modules/solver/agent/agent._cpp (hunk 1 above)
Comment (SergioMartin86) [reply source/modules/solver/agent/agent._cpp:181]: this is not a compute intensive call and some information is still useful to have (off policy ratio, beta factor), even if you don't use REFER

File: source/modules/solver/agent/continuous/GFPT/GFPT._cpp (hunk 2 above)
Comment (SergioMartin86) [reply source/modules/solver/agent/continuous/GFPT/GFPT._cpp:257]: because I forgot to remove it :P

**Review by wadaniel — APPROVED** [review]

**Pull request merged**